crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -182,6 +188,188 @@ impl TransactionBlock {
         .map(TransactionBlock::try_from)
         .transpose()
     }
+
+    pub(crate) async fn multi_query(
+        db: &Db,
+        digests: Vec<Digest>,
+    ) -> Result<BTreeMap<Digest, Self>, Error> {
+        use transactions::dsl;
+        let digests: Vec<_> = digests.into_iter().map(|d| d.to_vec()).collect();
+
+        let stored: Vec<StoredTransaction> = db
+            .results(move || {
+                dsl::transactions
+                    .filter(dsl::transaction_digest.eq_any(digests.clone()))
+                    .into_boxed()
+            })
+            .await
+            .map_err(|e| Error::Internal(format!("Failed to fetch transactions: {e}")))?;
+
+        let mut transactions = BTreeMap::new();
+        for tx in stored {
+            let digest = Digest::try_from(&tx.transaction_digest[..])
+                .map_err(|e| Error::Internal(format!("Bad digest for transaction: {e}")))?;
+
+            let transaction = TransactionBlock::try_from(tx)?;
+            transactions.insert(digest, transaction);
+        }
+
+        Ok(transactions)
+    }
+
+    pub(crate) async fn paginate(
+        db: &Db,
+        page: Page<u64>,
+        filter: TransactionBlockFilter,
+    ) -> Result<Connection<String, TransactionBlock>, Error> {
+        let (prev, next, results) = page
+            .paginate_query::<StoredTransaction, _, _, _>(db, move || {
+                use transactions as tx;
+                let mut query = tx::dsl::transactions.into_boxed();
+
+                if let Some(f) = &filter.function {
+                    let sub_query = tx_calls::dsl::tx_calls
+                        .select(tx_calls::dsl::tx_sequence_number)
+                        .into_boxed();
+
+                    query = query.filter(tx::dsl::tx_sequence_number.eq_any(f.apply(
```

Comment:
Just a curiosity: if `f` is malformed in a way, this whole query will return an error, though this `apply` call will be fine.

Reply:
Yes, if `f` is malformed we would detect that at a much earlier stage (when parsing the `TransactionBlockFilter`) so if you have an instance of `FqNameFilter` in your hands, you know for sure that you can apply it as a filter without problems.